src/components/Layout/Login.js

```diff
@@ -0,0 +1,106 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { withNamespaces } from 'react-i18next';
+
+import { Button, Divider, List, Popup } from 'semantic-ui-react';
+import { selectors, actions } from '../../redux/modules/auth';
+import { initKC } from '../../sagas/helpers/keycklockManager';
+import { DeviceInfoContext } from '../../helpers/app-contexts';
+import { getLanguageDirection } from '../../helpers/i18n-utils';
+import Link from '../Language/MultiLanguageLink';
+
+const Login = ({ t, language }) => {
+  const [isActive, setIsActive] = useState(false);
+  const { isMobileDevice }      = useContext(DeviceInfoContext);
+  const direction               = getLanguageDirection(language);
+  const popupStyle              = { direction };
+  const dispatch                = useDispatch();
+  const user                    = useSelector(state => selectors.getUser(state.auth));
+
+  useEffect(() => {
+    !user && initKC(dispatch, language);
```

Comment:
what happens when you change language before logging in?

Reply:
must to call init on first render only